constellation-app/constellation
# Prevent exception popup windows from being displayed when a notification was already shown.

### Prerequisites


    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Addresses issues raised in #1641 

When a plugin shows a notification to the user, it should no longer show an exception popup as well.

This is achieved by setting a block-popups flag on the Error Report that is generated when the exception is logged.

Note that this Pull Request only covers plugins that are part of the core "constellation" repo.
There will be a separate Pull Request for the plugins that are defined in the "constellation-adaptors" repo.



### Alternate Designs
None


### Why Should This Be In Core?

It should be a consistent standard across the product to have notification popups without an accompanying exception dialog.



### Benefits

Improved user experience.
No need to show an exception dialog as well as a simple notification when that notification alone will suffice.



### Possible Drawbacks
None



### Verification Process

Test these core plugins (from those listed in #1641):
* Import from GraphML file
* Import from GML file
* Import from Pajek file
* Extract types from text
* Utility > Select Top Node and the Import Options

Trying to run each plugin without completing the required details should cause a notification to be displayed.
For many cases, there should also be an Error Report generated and listed in the Error Report View.
For errors with an associated Error Report entry, confirm that the block-popups icon is showing on those entries.

Note that some plugins will throw an INFO severity level error.
The Error Report View will not show INFO popups by default. 
The default behaviour can be set in the Report Settings control.


### Applicable Issues

#1641
